### ISSUE TITLE
Add MapLibre to "Choosing an API/library"

### ISSUE THREAD

**Issue by louwers** [switch2osm/switch2osm#255]
Can MapLibre be added here, or would a PR be accepted that adds MapLibre?

https://switch2osm.org/using-tiles/

**Comment by Andygol** [comment]
@louwers do you mean something like this https://andygol.co.ua/switch2osm-mkdocs/using-tiles/vector-map-tiles/ + https://andygol.co.ua/switch2osm-mkdocs/serving-tiles/serving-vector-tiles/ ?

**Comment by SomeoneElseOSM** [comment]
> Can MapLibre be added here,

### The tl;dr answer:

Yes, but it needs to be more than a one-liner pointing at existing MapLibre resources

### In more detail:

I think rather more fundamentally than that the whole switch2osm site should acknowledge that vector tiles exist as a concept.  I suspect that https://switch2osm.org/using-tiles/ should be repurposed around "using _raster_ tiles" (which it still describes)**

What we probably need is a "vector tiles vs raster tiles" page above "Using Tiles" on the list at the right of https://switch2osm.org/using-tiles/ that

* explains what raster tiles are (some of which is at https://switch2osm.org/the-basics/ )
* explains what vector tiles are (something that fits in a bit before something like https://andygol.co.ua/switch2osm-mkdocs/using-tiles/vector-map-tiles/#Vector-Map-Tiles-Usage ) and that there are a huge variety of distribution mechanisms, file formats, content schemas, creation mechanisms etc.
* explains when you might want to use one and when the other.

I think that there then needs to be a page that explains the various providers and schemas in some detail, and after we definitely need a page for MapLibre GL JS and also one or two for MapLibre Native.

The challenge is that the documentation on switch2osm.org is aimed at someone who has enough knowledge to "have maps on their website" but probably no more than that. A recent comment about OSM software (not about MapLibre) said something along the lines of "I read as far as `npm install` and then realised it was not for me", so something like https://maplibre.org/maplibre-gl-js/docs/ is a bit of an issue as it stands.  However, it would absolutely be possible to create introductory-level worked examples that used the OSMF demo tiles at https://community.openstreetmap.org/t/vector-tiles-on-osmf-hardware/121501 (or any more feature-rich successor to those).

Challenges with creating that would include:
* Even with a very light schema, a "relatively complete"  style that uses it such as https://github.com/SomeoneElseOSM/SomeoneElse-vector-web-display/blob/main/resources/svwd03_style.json is quite large *7k lines).  An introductory example might want to concentrate on only countries and borders and some very limited features.
* Users are going to want to create their own graphics, and they'll need to create sprites from those.  When I tried to do this I couldn't find any good options here, so I ended up creating my own scripts for this, but I'd be wary about writing switch2osm pages suggesting using those unless there really isn't a better option.
* Users are probably going to want to choose their own fonts, rather than rely on a small subset of existing ones such as https://github.com/klokantech/klokantech-gl-fonts .  The challenge with that 

For that to be hosted on a web page, the required HTML / JS is pretty minimal: https://github.com/SomeoneElseOSM/SomeoneElse-vector-web-display/blob/main/resources/svwd03_index.html .

For MapLibre Native Android, https://maplibre.org/maplibre-native/android/examples/getting-started/ is much closer to what' needed, although it could benefit from a bit more text at the top to guide people who haven't seen Android Studio before through some of the terminology.

** a disclaimer here is that I've not had much to do with the "using tiles" side of the site, so this is really a view from the outside looking in.
